### PR TITLE
feat: append id of the AnoymousUserId instance to client authorization.

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description

The requirement is to generate an id shorter than 25 characters  that allows to associate a user with a course, the current CourseEnrollment implementation does that, however if the record is deleted the association will be lost, so this includes the id of the AnonymousUserId instance, so if the course enrollment record is deleted the id could help to identify the user-course association 

Basically this modifies the method `generate_client_authorization_id` in order to append the AnonymousUserId instance id to the current value.

## Additional information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
- Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.

## Checklist for Merge

- [ ] Tested in a remote environment
- [ ] Updated documentation
- [ ] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

File: eox_nelp/pearson_vue/utils.py [hunk 1]
```diff
@@ -40,5 +40,7 @@ def generate_client_authorization_id(user_id: int, course_id: str) -> str:
         str: string that represents the client_auth_id
     """
     course_enrollment = CourseEnrollment.objects.get(user_id=user_id, course_id=course_id)
+    anonymous_user_id = anonymous_id_for_user(course_enrollment.user, course_id)
+    anonymous_user_id_instance = AnonymousUserId.objects.get(anonymous_user_id=anonymous_user_id)
```
Comment (johanseto) [inline eox_nelp/pearson_vue/utils.py:44]: Why not simply use the AnonymousUserId alone for identification? I believe it's unique per course and user. However, if some AnonymousUserId values are missing, then perhaps a combination of course_enrollment_id and anonymous_user_id could be used for tracking. That is is the reason?

Comment (andrey-canon) [reply eox_nelp/pearson_vue/utils.py:44]: Same CourseEnrollment reason, if someone deletes the record we won't be able to track the user

File: eox_nelp/pearson_vue/utils.py [hunk 2]
```diff
@@ -40,5 +40,7 @@ def generate_client_authorization_id(user_id: int, course_id: str) -> str:
...
 
-    return str(course_enrollment.id + CLIENT_AUTHORIZATION_ID_OFFSET)
+    return f"{course_enrollment.id + CLIENT_AUTHORIZATION_ID_OFFSET}-{anonymous_user_id_instance.id}"
```
Comment (johanseto) [inline eox_nelp/pearson_vue/utils.py:46]: I think we could remove the `CLIENT_AUTHORIZATION_ID_OFFSET`.

Comment (andrey-canon) [reply eox_nelp/pearson_vue/utils.py:46]: I thought you like it :)

**Review by johanseto — APPROVED** [review]

**Pull request merged**